### ISSUE TITLE
Potential utility to install package icons

### ISSUE THREAD

**Issue by GetPsyched** [NixOS/nixpkgs#278256]
Having recently learnt about [`installShellFiles` hook](https://github.com/NixOS/nixpkgs/blob/f459921fa4f1990d81b7e5fa768b08b66bde3c9f/doc/hooks/installShellFiles.section.md), I think there should be one for installing icons as well.

Since a lot of derivations install `*.desktop` files (using the convenient `copyDesktopItems` and `makeDesktopItem` combo), they also need to install the corresponding icons; and this is done currently with either `cp` or `install` into `$out/share/icons/hicolor/*/apps`.

I think this can be simplified by making available an `installIcon` command, much like `installManPage`. Just wanted to get comments on this; if it gets somewhere, I could PR the concluded changes.

**Comment by SuperSandro2000** [comment]
Since there is a lot which can go wrong here especially with icon sizes and sometimes required testing on wayland DEs this should probably done.

I think we should first try to generalize the most common patterns and convert those and leave the tricky cases left to manual installing the files

**Comment by GetPsyched** [comment]
Yeah, I was thinking of the resizing issue since some apps give icons incompatible with hicolor theme sizes. Perhaps the utility could also abstract out resizing to save inconsistencies among all the packages (using different tools to resize such as `imagemagick`, etc.)

Perhaps a `makeIcon` function (similar to `makeDesktopItem`) which takes in a path and some meta data about it like the resolution and mime type. Then it can internally decide where to put it and if it needs resizing or not. This could also be achieved by a shell hook I mentioned originally.
